src/ParallelAlgorithms/Interpolation/Events/InterpolateWithoutInterpComponent.hpp

```diff
@@ -124,32 +125,52 @@ class InterpolateWithoutInterpComponent<VolumeDim, InterpolationTargetTag,
     Variables<typename InterpolationTargetTag::vars_to_interpolate_to_target>
         interp_vars(mesh.number_of_grid_points());
 
-    // Clang-tidy wants extra braces for `if constexpr`
-    if constexpr (std::is_same_v<tmpl::list<>, typename InterpolationTargetTag::
-                                                   compute_items_on_source>) {
-      // 1.a Copy the tensors directly into the variables; no need to
-      // make a DataBox because we have no ComputeItems.
+    if constexpr (InterpolationTarget_detail::has_compute_vars_to_interpolate_v<
+                      InterpolationTargetTag>) {
+      // 1a. Call compute_vars_to_interpolate.  Need the source in a
+      // Variables, so copy the variables here.
+      // This copy would be unnecessary if we passed a Variables into
+      // InterpolateWithoutInterpComponent instead of passing
+      // individual Tensors, which would require that this Variables is
+      // something in the DataBox. (Note that
+      // InterpolationTarget_detail::compute_dest_vars_from_source_vars
+      // allows the source variables to be different from the
+      // destination variables).
+      Variables<db::AddSimpleTags<Tensors...>> source_vars(
+          mesh.number_of_grid_points());
+      [[maybe_unused]] const auto copy_to_variables =
+          [&source_vars](const auto tensor_tag_v, const auto& tensor) {
+            using tensor_tag = tmpl::type_from<decltype(tensor_tag_v)>;
+            get<tensor_tag>(source_vars) = tensor;
+            return 0;
+          };
+      expand_pack(copy_to_variables(tmpl::type_<Tensors>{}, tensors)...);
+
+      InterpolationTarget_detail::compute_dest_vars_from_source_vars<
+          InterpolationTargetTag>(
+          make_not_null(&interp_vars), source_vars,
+          get<domain::Tags::Domain<Metavariables::volume_dim>>(cache), mesh,
+          element_ids[0], cache, temporal_id);
```

Comment:
`element_ids[0]` -> `array_index` (I got confused why `element_ids` is a list here. You can rename `array_index` to `element_id` if you prefer)